include/ed/models/shape_loader.h

```diff
@@ -0,0 +1,169 @@
+#ifndef ED_SHAPE_LOADER_H_
+#define ED_SHAPE_LOADER_H_
+
+#include <geolib/datatypes.h>
+#include <geolib/Mesh.h>
+#include <geolib/Shape.h>
+
+namespace ed
+{
+
+namespace models
+{
+
+
+/**
+ * @brief createCylinder create a mesh from radius and height
+ * @param shape filled mesh
+ * @param radius radius of the cylinder
+ * @param height height of the cylinder
+ * @param num_corners divided the circumference in N points and N+1 lines
+ */
+void createCylinder(geo::Shape& shape, double radius, double height, int num_corners = 12)
+{
+    geo::Mesh mesh;
+
+    // Calculate vertices
+    for(int i = 0; i < num_corners; ++i)
+    {
+        double a = 6.283 * i / num_corners;
+        double x = sin(a) * radius;
+        double y = cos(a) * radius;
+
+        mesh.addPoint(x, y, -height / 2);
+        mesh.addPoint(x, y,  height / 2);
+    }
+
+    // Calculate top and bottom triangles
+    for(int i = 1; i < num_corners - 1; ++i)
+    {
+        int i2 = 2 * i;
+
+        // bottom
+        mesh.addTriangle(0, i2, i2 + 2);
+
+        // top
+        mesh.addTriangle(1, i2 + 3, i2 + 1);
+    }
+
+    // Calculate side triangles
+    for(int i = 0; i < num_corners; ++i)
+    {
+        int j = (i + 1) % num_corners;
+        mesh.addTriangle(i * 2, i * 2 + 1, j * 2);
+        mesh.addTriangle(i * 2 + 1, j * 2 + 1, j * 2);
+    }
+
+    shape.setMesh(mesh);
+}
+
+int getMiddlePoint(geo::Mesh& mesh, int i1, int i2, std::map<long, int> cache, double radius)
```

Comment:
Needs doc string. Also, since the cache and the mesh seem tied together, wouldn't it make sense to store them together in an object? (rather than this function, with separate mesh and cache somewhere)

Reply:
This code has only be moved. And it has been copied from stackoverflow.